web/src/utils/helpers.ts

```diff
@@ -60,13 +66,68 @@ export function getProductionCo2Intensity(
   return dischargeCo2Intensity;
 }
 
-/**
- * Returns a link which maintains search and hash parameters
- * @param to
- */
+export function useNavigateWithParameters() {
+  const navigator = useNavigate();
+  const location = useLocation();
+  const {
+    zoneId: previousZoneId,
+    urlTimeAverage: previousTimeAverage,
+    urlDatetime: previousDatetime,
+  } = useParams();
+  console.log('previousDatetime', previousDatetime);
```

Comment:
nit: got a console log here and on line 105

Reply:
Updated!